src/Json.Schema.ToDotNet.UnitTests/Hints/ClassNameHintTests.cs

```diff
@@ -269,7 +269,208 @@ public int GetHashCode(FileData obj)
+    /// </summary>
+    [GeneratedCode(""Microsoft.Json.Schema.ToDotNet"", """ + VersionConstants.FileVersion + @""")]
+    internal sealed class CEqualityComparer : IEqualityComparer<C>
+    {
+        internal static readonly CEqualityComparer Instance = new CEqualityComparer();
+
+        public bool Equals(C left, C right)
+        {
+            if (ReferenceEquals(left, right))
+            {
+                return true;
+            }
+
+            if (ReferenceEquals(left, null) || ReferenceEquals(right, null))
+            {
+                return false;
+            }
+
+            if (!FileData.ValueComparer.Equals(left.File, right.File))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public int GetHashCode(C obj)
+        {
+            if (ReferenceEquals(obj, null))
+            {
+                return 0;
+            }
+
+            int result = 17;
+            unchecked
+            {
+                if (obj.File != null)
+                {
+                    result = (result * 31) + obj.File.ValueGetHashCode();
+                }
+            }
+
+            return result;
+        }
+    }
+}",
+
+// HintedClassText
+@"using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace N
+{
+    [DataContract]
+    [GeneratedCode(""Microsoft.Json.Schema.ToDotNet"", """ + VersionConstants.FileVersion + @""")]
+    public partial class FileData : PropertyBagHolder
+    {
+        public static IEqualityComparer<FileData> ValueComparer => FileDataEqualityComparer.Instance;
+
+        public bool ValueEquals(FileData other) => ValueComparer.Equals(this, other);
+        public int ValueGetHashCode() => ValueComparer.GetHashCode(this);
+
+        [DataMember(Name = ""path"", IsRequired = false, EmitDefaultValue = false)]
+        public string Path { get; set; }
+    }
+}",
+
+// HintedClassComparerText
+@"using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+
+namespace N
+{
+    /// <summary>
+    /// Defines methods to support the comparison of objects of type FileData for equality.
+    /// </summary>
+    [GeneratedCode(""Microsoft.Json.Schema.ToDotNet"", """ + VersionConstants.FileVersion + @""")]
+    internal sealed class FileDataEqualityComparer : IEqualityComparer<FileData>
+    {
+        internal static readonly FileDataEqualityComparer Instance = new FileDataEqualityComparer();
+
+        public bool Equals(FileData left, FileData right)
+        {
+            if (ReferenceEquals(left, right))
+            {
+                return true;
+            }
+
+            if (ReferenceEquals(left, null) || ReferenceEquals(right, null))
+            {
+                return false;
+            }
+
+            if (left.Path != right.Path)
+            {
+                return false;
+            }
+
+            return true;
+        }
```

Comment:
>} [](start = 8, length = 1)

Now this is a separate, existing, bug, which I will file. The generated equality comparer doesn't compare the base classes!

Reply:
@michaelcfanning [[https://github.com/Microsoft/jschema/issues/55](https://github.com/Microsoft/jschema/issues/55)](https://github.com/Microsoft/jschema/issues/55), "Equality comparer doesn't compare base types".

---
In reply to: [226104727](https://github.com/Microsoft/jschema/pull/54#discussion_r226104727) [](ancestors = 226104727)